Fix/correct f rate value and global system disable fix

Kasmadei opened the pull request:
Fixes 2 points. 
 1) Incorrect value of manual f. rate after server update.
 2) https://github.com/kbss-cvut/fta-fmea-ui/issues/415

LaChope commented on src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx:
```diff
@@ -83,21 +83,25 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
         ...shapeToolData,
         selectedEstimate: { iri: snsOperationalIri, value: snsOperationalFailureRate },
-        probability: snsOperationalFailureRate,
+        probability: Number(snsOperationalFailureRate),
```
If possible, it would be better to give the type to `snsOperationalFailureRate` than casting it.

Kasmadei replied:
Fixed.

LaChope commented on src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx:
```diff
@@ -83,21 +83,25 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
-    onEventUpdated({ ...shapeToolData, probability: snsManuallyDefinedFailureRate });
+  const handleManualFailureRateUpdate = (eventType: EventType) => {
+    if (eventType === EventType.EXTERNAL) {
+      onEventUpdated({ ...shapeToolData, probability: Number(externalManuallyDefinedFailureRate) });
```
Same as above

Kasmadei replied:
Fixed.

LaChope commented on src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx:
```diff
@@ -83,21 +83,25 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
+    if (eventType === EventType.EXTERNAL) {
+      onEventUpdated({ ...shapeToolData, probability: Number(externalManuallyDefinedFailureRate) });
+    } else {
+      onEventUpdated({ ...shapeToolData, probability: Number(snsManuallyDefinedFailureRate) });
```
same as above

Kasmadei replied:
Fixed.

LaChope reviewed (changes requested):
See my comments

blcham commented on src/components/appBar/AppBar.tsx:
```diff
@@ -52,7 +52,7 @@ const AppBar = ({ title, showBackButton = false, topPanelHeight }: Props) => {
   const { appBarTitle, systemsList } = useAppBar();
   const location = useLocation();
-  const dropdownDisabled = shouldDropdownBeDisabled(location.pathname);
+  const disabled = shouldSystemChangeBeDisabled(location.pathname);
```
@Kasmadei renamed the property because it does not do only "disable dropdown" but also "disable removal of value"

How about `isGlobalSystemSwitchDisabled` ?

LaChope replied:
Agree

Kasmadei replied:
Changed.

blcham reviewed (commented):
Can be merged after mentioned issues are resolved.

Kasmadei commented:
@blcham @LaChope Hi, i have pushed fixes, but i can't test it :( Unfortunately, I can't fetch our main tree example. I got a 500 status error, even after updating Docker. Do u have same problem, could you check, please ? 

<img width="1440" alt="Screenshot 2024-06-19 at 8 38 31 PM" src="https://github.com/kbss-cvut/fta-fmea-ui/assets/49252228/ddecad42-19b9-4f32-ae72-9ab40094352e">

LaChope commented:
I cannot reproduce the issue (from main branch - I do not have access to your branch), can you share the logs from the backend?

LaChope commented on src/components/editor/faultTree/menu/faultEvent/FaultEventMenu.tsx:
```diff
@@ -83,24 +83,24 @@ const FaultEventMenu = ({ shapeToolData, onEventUpdated, refreshTree, rootIri }:
         ...shapeToolData,
         selectedEstimate: { iri: snsPredictedIri, value: snsPredictedFailureRate },
-        probability: Number(snsPredictedFailureRate),
+        probability: snsPredictedFailureRate,
```
Not sure to understand, there was no need to cast, or now there is a typescript warning?

Kasmadei replied:
Fixed.

blcham commented:
@Kasmadei I cannot reproduce it either, please provide some logs

Kasmadei commented:
@LaChope @blcham Hi, I attempted to resolve the issue with the following Docker steps:

docker-compose down --rmi all --volumes
docker system prune -a --volumes
docker-compose pull
docker-compose build --no-cache
docker-compose up
Afterward, I navigated to http://localhost:1235/services/auth > Administration Console > selected fta-fmea in the dropdown > Users > Created a new user. I also added a password in credentials and assigned a few roles in Role mapping.

I added a login and password to keycloak-auth/.env.

I also created a .env.development file with the following content:

``` 
FTA_FMEA_BASENAME=''
FTA_FMEA_API_URL=http://localhost:1235/services/fta-fmea-server
FTA_FMEA_ADMIN_REGISTRATION_ONLY=false
FTA_FMEA_TITLE='Development FTA/FMEA Tool'
FTA_FMEA_AUTHENTICATION='oidc'
FTA_FMEA_AUTH_SERVER_URL=http://localhost:1235/services/auth/realms/fta-fmea
FTA_FMEA_AUTH_CLIENT_ID='fta-fmea'
```

Finally, I went to http://localhost:5173/ and after successfully logging in, I got stuck on the "Redirecting..." screen. Did i forget something ? 
<img width="1438" alt="Screenshot 2024-06-20 at 8 05 13 PM" src="https://github.com/kbss-cvut/fta-fmea-ui/assets/49252228/cff5270f-316b-4702-a7f7-cfe67dd60fb8">

Kasmadei commented:
@LaChope @blcham Hi, I attempted to resolve the issue with the following Docker steps:

docker-compose down --rmi all --volumes
docker system prune -a --volumes
docker-compose pull
docker-compose build --no-cache
docker-compose up
Afterward, I navigated to http://localhost:1235/services/auth > Administration Console > selected fta-fmea in the dropdown > Users > Created a new user. I also added a password in credentials and assigned a few roles in Role mapping.

I added a login and password to keycloak-auth/.env.

I also created a .env.development file with the following content:

``` 
FTA_FMEA_BASENAME=''
FTA_FMEA_API_URL=http://localhost:1235/services/fta-fmea-server
FTA_FMEA_ADMIN_REGISTRATION_ONLY=false
FTA_FMEA_TITLE='Development FTA/FMEA Tool'
FTA_FMEA_AUTHENTICATION='oidc'
FTA_FMEA_AUTH_SERVER_URL=http://localhost:1235/services/auth/realms/fta-fmea
FTA_FMEA_AUTH_CLIENT_ID='fta-fmea'
```

Finally, I went to http://localhost:5173/ and after successfully logging in, I got stuck on the "Redirecting..." screen. Did i forget something ? 
<img width="1438" alt="Screenshot 2024-06-20 at 8 05 13 PM" src="https://github.com/kbss-cvut/fta-fmea-ui/assets/49252228/cff5270f-316b-4702-a7f7-cfe67dd60fb8">

blcham commented:
@Kasmadei Yes you forgot to set up 2 values in keycloak:
- web-origin
- some other value in the same form

Kasmadei commented:
Thanks, it works 👍
